src/services/UserInfoRequestProcessor.ts

```diff
@@ -0,0 +1,189 @@
+  	this.personIdentityTableName = checkEnvironmentVariable(EnvironmentVariables.PERSON_IDENTITY_TABLE_NAME, this.logger);
+
+  	this.BavService = BavService.getInstance(sessionTableName, this.logger, createDynamoDbClient());
+  	this.kmsDecryptor = new KmsJwtAdapter(signinKeyIds);
+		this.verifiableCredentialService = VerifiableCredentialService.getInstance(sessionTableName, this.kmsDecryptor, this.issuer, this.logger);
+	}
+
+	static getInstance(logger: Logger, metrics: Metrics): UserInfoRequestProcessor {
+  	if (!UserInfoRequestProcessor.instance) {
+  		UserInfoRequestProcessor.instance = new UserInfoRequestProcessor(logger, metrics);
+  	}
+  	return UserInfoRequestProcessor.instance;
+	}
+
+	async processRequest(event: APIGatewayProxyEvent): Promise<Response> {
+		// Validate the Authentication header and retrieve the sub (sessionId) from the JWT token.
+		let sessionId: string;
+		try {
+			sessionId = await eventToSubjectIdentifier(this.kmsDecryptor, event);
+		} catch (error) {
+			if (error instanceof AppError) {
+				this.logger.error("Error validating Authentication Access token from headers: ", {
+					error,
+					messageCode: MessageCodes.INVALID_AUTH_CODE,
+				});
+				return new Response(HttpCodesEnum.UNAUTHORIZED, "Unauthorized");
+			}
+			this.logger.error("Unexpected error occurred", {
+				error,
+				messageCode: MessageCodes.SERVER_ERROR,
+			});
+			return new Response(HttpCodesEnum.SERVER_ERROR, "Server Error");
+		}
+
+		// add sessionId to all subsequent log messages
+		this.logger.appendKeys({ sessionId });
+		
+		const session: ISessionItem | undefined = await this.BavService.getSessionById(sessionId);
+		if (!session) {
+			this.logger.error("No session found", {
+				messageCode: MessageCodes.SESSION_NOT_FOUND,
+			});
+			return new Response(HttpCodesEnum.UNAUTHORIZED, "Unauthorized");
+		}
+
+		// add govuk_signin_journey_id to all subsequent log messages
+		this.logger.appendKeys({ govuk_signin_journey_id: session.clientSessionId });
+
+		this.logger.info("Found Session:", {
+			// note: we only log specific non-PII attributes from the session object:
+			session: {
+				authSessionState: session.authSessionState,
+				accessTokenExpiryDate: session.accessTokenExpiryDate,
+				attemptCount: session.attemptCount,
+				authorizationCodeExpiryDate: session.authorizationCodeExpiryDate,
+				createdDate: session.createdDate,
+				expiryDate: session.expiryDate,
+				redirectUri: session.redirectUri,
+			},
+		});
+  
+		this.metrics.addMetric("found session data", MetricUnits.Count, 1);
+
+		const personInfo: PersonIdentityItem | undefined = await this.BavService.getPersonIdentityBySessionId(sessionId, this.personIdentityTableName);
+		if (!personInfo) {
+			this.logger.error("No person found with this session ID", {
+				messageCode: MessageCodes.PERSON_NOT_FOUND,
+			});
+			return new Response(HttpCodesEnum.UNAUTHORIZED, "Unauthorized");
```

Comment:
do you think it would be helpful to return a more descriptive response message?

Reply:
Do we want to have a more descriptive response message? We have it in the log for API response I thought the more vague the better?

Reply:
Interesting one, I guess would a more descriptive response be useful to someone who wouldn't look at the logs, like IPV?

Reply:
API Responses updated